tests/AutoReview/FixerTest.php

```diff
@@ -74,23 +96,64 @@ public function testFixerDefinitions(FixerInterface $fixer)
                 $sample instanceof FileSpecificCodeSampleInterface ? $sample->getSplFileInfo() : $dummyFileInfo,
                 $tokens
             );
-            $this->assertTrue($tokens->isChanged(), sprintf('[%s] Sample #%d is not changed during fixing.', $fixer->getName(), $sampleCounter));
+
+            $this->assertTrue($tokens->isChanged(), sprintf('[%s] Sample #%d is not changed during fixing.', $fixerName, $sampleCounter));
 
             $duplicatedCodeSample = array_search(
                 $sample,
                 array_slice($samples, 0, $sampleCounter),
                 false
             );
+
             $this->assertFalse(
                 $duplicatedCodeSample,
-                sprintf('[%s] Code sample #%d duplicates #%d.', $fixer->getName(), $sampleCounter, $duplicatedCodeSample)
+                sprintf('[%s] Sample #%d duplicates #%d.', $fixerName, $sampleCounter, $duplicatedCodeSample)
             );
         }
 
+        if ($fixerIsConfigurable) {
+            if (isset($configSamplesProvided['default'])) {
+                reset($configSamplesProvided);
+                $this->assertSame('default', key($configSamplesProvided), sprintf('[%s] First sample must be for the default configuration.', $fixerName));
+            } else {
+                $this->assertArrayHasKey($fixerName, $this->allowedRequiredOptions, sprintf('[%s] Has no sample for default configuration.', $fixerName));
+            }
+
+            // test that all configuration options are used at least one time in the samples
+
+            /** @var FixerOption[] $options */
+            $options = $fixer->getConfigurationDefinition()->getOptions();
+            $optionsNames = array();
+
+            if (count($options)) {
+                foreach ($options as $option) {
+                    $optionsNames[$option->getName()] = 0; //$option->hasDefault() ? 1 : 0;
```

Comment:
after the PR of @keradus we are sure at this point in the code a fixer has a code sample with default configuration. Here I start counting that _besides_ that code sample with default config there is at least one other code sample using the option, therefor we always have a sample with the option _other_ than the default.
(we still might have duplicates though, I address that later I hope, but configuration arrays are not easy to compare)